Makefile

```diff
@@ -103,6 +105,11 @@ table-check:
 	docker run --rm -v $(PWD)/model:/source -v $(PWD)/docs:/spec \
 		otel/semconvgen:$(SEMCONVGEN_VERSION) -f /source markdown -md /spec --md-check --md-use-badges --md-stable
 
+.PHONY: compatibility-check
+compatibility-check:
```

Comment:
TODO:

based on https://github.com/open-telemetry/semantic-conventions/pull/770#discussion_r1502392874, I will also update contributing guide to add instructions on how to run it and how to deprecate attributes/metrics